Firmware/ultralcd.cpp

```diff
@@ -5191,7 +5206,13 @@ static void lcd_main_menu()
     } else if (!Stopped) {
         MENU_ITEM_SUBMENU_P(_i("Preheat"), lcd_preheat_menu);////MSG_PREHEAT c=18
     }
-
+    if (GetPrinterState() < PrinterState::IsSDPrinting && M79_timer_get_status()) {
```

Comment:
Based on this statement, the order / numerical values of PrinterState members is important. A big warning should be added to the structure so that no one tries to reorder the items arbitrarily.